framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4923,6 +5003,73 @@ VkResult VulkanReplayConsumerBase::OverrideCreatePipelineCache(
+            capture_pipeline_cache_data_size_ = create_info.initialDataSize;
+
+            object_info_table_.VisitPipelineCacheInfo([this](const PipelineCacheInfo* pipeline_cache_info) {
+                GFXRECON_ASSERT(pipeline_cache_info != nullptr);
+
+                auto iterator = pipeline_cache_info->pipeline_cache_data.find(capture_pipeline_cache_data_hash_);
+
+                if (iterator != pipeline_cache_info->pipeline_cache_data.end())
+                {
+                    // We found pipeline cache data vector which has same hash value, will continue to check if it has
+                    // same capture time pipeline cache data.
+                    auto& cache_data = iterator->second;
+
+                    for (auto& existing_cache_data : cache_data)
+                    {
+                        if (capture_pipeline_cache_data_size_ == existing_cache_data.capture_cache_data.size())
+                        {
+                            // Target pipeline cache data has same size, we continue to check if it also has same data.
+                            if (memcmp(existing_cache_data.capture_cache_data.data(),
+                                       capture_pipeline_cache_data_,
+                                       capture_pipeline_cache_data_size_) == 0)
+                            {
+                                // Now we found the pipeline cache data, here we record its replay time data because we
+                                // need this data to replace capture time cache data in the vkCreatePipelineCache call.
+                                matched_replay_cache_data_exist_ = true;
+                                matched_replay_cache_data_.resize(existing_cache_data.replay_cache_data.size());
+                                memcpy(matched_replay_cache_data_.data(),
+                                       existing_cache_data.replay_cache_data.data(),
+                                       existing_cache_data.replay_cache_data.size());
+                                break;
+                            }
+                        }
+                    }
+                }
+            });
+
+            if (matched_replay_cache_data_exist_)
+            {
+                VkPipelineCacheCreateInfo override_create_info = (*replay_create_info);
+                override_create_info.initialDataSize           = matched_replay_cache_data_.size();
+                override_create_info.pInitialData              = matched_replay_cache_data_.data();
+
+                return func(device_info->handle,
+                            &override_create_info,
+                            GetAllocationCallbacks(pAllocator),
+                            pPipelineCache->GetHandlePointer());
+            }
+            else
+            {
+                GFXRECON_LOG_WARNING(
+                    "There's initial pipeline cache data in VkPipelineCacheCreateInfo, but no corresponding "
```

Comment:
I feel this is normal operation for any app that loads a previous cache from disk, but here's a PR to make it a DEBUG message.  https://github.com/LunarG/gfxreconstruct/pull/1527

Reply:
Yes, once the warning shows up, it means if playback on different hardware/driver, playback performance may have difference with capture time because of extra compile in pipeline creation.
Because if an application load cache data from file, that mean these cache data come from the title previous run, we cannot get its corresponding replay time cache data. When utilizing the cache data, two scenarios typically arise:
1. The cache data is compatible with current hardware/driver, the cache data will be used.
2. The cache data is not compatible with current hardware/driver, the cache data will be ignored by driver, the following pipeline creations with the cache object may trigger compile.
As for the second case, replay performance will have some difference with capture time because of the extra compile. of course, for any case, it doesn't cause playback failure, I agree change to GFXRECON_LOG_DEBUG or GFXRECON_LOG_WARNING_ONCE.

Reply:
So it's going to behave as if the user changed the GPU in the machine and ran the original app?